data-structures/hash-tables.md

```diff
@@ -39,11 +40,26 @@ Knowing how hash tables work will give you a deeper understanding of why hash ta
 
 ### Common Mistakes / Misconceptions
 
-What is the difference between a hash map and a hash table? The two are used interchangeably.
+- What is the difference between a hash map and a hash table? The two are used interchangeably.
+
+- When should I use an array instead of a hash table? If your keys are sequential integers.
+
+- When does a JavaScript object stop being a hash table? When a property is added as a function.
+
+- [hashmap implementation](https://gist.github.com/dtx/3791420)
+
+ ![Image result for difference between hashmap and hashtable](https://4.bp.blogspot.com/-K0P34lM0LbI/VR4yDT398XI/AAAAAAAAGFQ/uLwC_jq9hvU/s1600/Java%2B-%2BCollection%2BFramework%2B-%2BHashMap%2BVs.%2BHashtable.jpg)
```

Comment:
I love charts, but this needs to be in JS too. Half of the examples aren’t helpful for JS students.